TLM/TLM/UI/SubTools/PrioritySigns/Overlay.cs

```diff
@@ -0,0 +1,516 @@
+        /// <param name="node">Junction node ref.</param>
+        /// <param name="camPos">Camera position.</param>
+        /// <param name="stateUpdated">?</param>
+        /// <returns>Whether any of the signs was hovered.</returns>
+        public bool DrawSignHandles(ushort nodeId,
+                                    ref NetNode node,
+                                    ref Vector3 camPos,
+                                    out bool stateUpdated) {
+            bool isAnyHovered = false;
+            stateUpdated = false;
+
+            // Quit now if:
+            //   * view only,
+            //   * and no permanent overlay enabled,
+            //   * and is not Prio Signs tool
+            if (this.ViewOnly &&
+                !(Options.junctionRestrictionsOverlay ||
+                  MassEditOverlay.IsActive) &&
+                this.mainTool_.GetToolMode() != ToolMode.JunctionRestrictions) {
+                return false;
+            }
+
+            // NetManager netManager = Singleton<NetManager>.instance;
+            Color guiColor = GUI.color;
+            Vector3 nodePos = Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId].m_position;
+            IExtSegmentEndManager segEndMan = Constants.ManagerFactory.ExtSegmentEndManager;
+
+            for (int i = 0; i < 8; ++i) {
+                ushort segmentId = node.GetSegment(i);
+
+                if (segmentId == 0) {
+                    continue;
+                }
+
+                bool isStartNode =
+                    (bool)Constants.ServiceFactory.NetService.IsStartNode(segmentId, nodeId);
+
+                bool isIncoming = segEndMan
+                                  .ExtSegmentEnds[segEndMan.GetIndex(segmentId, isStartNode)]
+                                  .incoming;
+
+                NetInfo segmentInfo = Singleton<NetManager>
+                                      .instance
+                                      .m_segments
+                                      .m_buffer[segmentId]
+                                      .Info;
+
+                ItemClass connectionClass = segmentInfo.GetConnectionClass();
+
+                if (connectionClass.m_service != ItemClass.Service.Road) {
+                    continue; // only for road junctions
+                }
+
+                //------------------------------------
+                // Draw all junction restriction signs
+                // Determine direction from node center towards each segment center and use that
+                // as axis Y, and then dot product gives "horizontal" axis X
+                //------------------------------------
+                Vector3 segmentCenterPos = Singleton<NetManager>
+                                           .instance
+                                           .m_segments
+                                           .m_buffer[segmentId]
+                                           .m_bounds
+                                           .center;
```

Comment:
as I said before this is better be :
```
//psudo-code:
pointA, tanA = segment.CalculateCorner(junctionID, left side )
pointB, tanB = segment.CalculateCorner(junctionID, right side)
segmentCenterPos = (postA+pointB)/2
yBasis=(tanA+tanB ).normalized
```

Reply:
using `CalucluateCorner` would replace some of the calculations performed in `SignsLayout`.